plugins/pubIds/doi/DOIPubIdPlugin.inc.php

```diff
@@ -500,4 +502,47 @@ public function addPublishFormNotice($hookName, $form)
             ]));
         }
     }
+
+    /**
+    * Add DOI to citation data used by the CitationStyleLanguage plugin
+    *
+    * @see CitationStyleLanguagePlugin::getCitation()
+    *
+    * @param $hookname string CitationStyleLanguage::citation
+    * @param $args array
+    */
+    public function getCitationData($hookname, $args): void
+    {
+        /** @var stdClass $citationData */
+        $citationData = $args[0];
+        /** @var Submission $submission */
+        $submission = $args[2];
+        /** @var Publication $publication */
+        $publication = $args[5] ?? $submission->getCurrentPublication();
+
+        if ($citationData->type === 'chapter') {
+            $chapter = null;
+            $chapters = $publication->getData('chapters');
+            /** @var Chapter $ichapter */
+            foreach ($chapters as $ichapter) {
+                if ((int) $ichapter->getSourceChapterId() === (int) $citationData->id) {
+                    $chapter = $ichapter;
+                    break;
+                }
+            }
```

Comment:
Instead of searching for the chapter again here, add it to the data passed to the hook at `HookRegistry::call('CitationStyleLanguage::citation',`.